### ISSUE TITLE
Support for accessibility

### ISSUE THREAD

**Issue by yousifalraheem** [sebgroup/ng-components#29]
### Description
When working in a project that requires support for accessibility features like a screen reader, these components should have all the necessary flags like `aria-` flags to support accessibility features. Form components should be focusable with pressing `tab` and collapsable items should declare its state and the new content of the changed area. The HTML markup should be used semantically so that screen readers can identify what the user is hovering over.

I believe this feature must be incorporated from the start instead of leaving it to the end since some of these changes might cause a major restructuring if it were to be implemented later.

### References
https://developer.mozilla.org/en-US/docs/Web/Accessibility
https://www.w3.org/WAI/standards-guidelines/wcag/
